private/byte.rkt

```diff
@@ -262,3 +263,17 @@
       (for ([b (in-byte x)]
             [i (in-range 8)])
         (check-equal? b (byte-ref x i))))))
+
+(define (byte-hamming-weight b)
+  (apply + (in-byte b)))
```

Comment:
The `apply` function only works with lists (unfortunately), and the `in-byte` sequence constructor doesn't promise that the sequence it returns is always a list. You can use `(for/sum ([bit (in-byte b)]) bit)` instead to make it work with any sequence.

Reply:
Note: I opened racket/racket#2845 to request a `sequence-apply` function in `racket/sequence`